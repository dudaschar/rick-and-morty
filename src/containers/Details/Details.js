import React from 'react';
import PropTypes from 'prop-types';

import Button from 'src/components/Button/Button';

import { Wrapper, CharacterDetails, Card, ImageDetail } from './styles/Wrapper';
import { CharacterName, EpisodesTitle } from './styles/Typography';
import { List, ListItem } from './styles/List';
import { Image } from './styles/Image';

function Details({ data }) {
  return (
    <Wrapper>
      <Card>
        <CharacterDetails>
          <CharacterName>{data.name}</CharacterName>
          <p><strong>Status:</strong> {data.status}</p>
          <p><strong>Species:</strong> {data.species}</p>
          <p><strong>Gender:</strong> {data.gender}</p>
          <p><strong>Type:</strong> {data.type !== '' ? data.type : 'N/A'}</p>
          <p><strong>Origin:</strong> {data.origin.name}</p>
          <p><strong>Location:</strong> {data.location.name}</p>
          <EpisodesTitle>Episodes</EpisodesTitle>
          {data.episode.map(ep => (
            <List key={ep.name}>
              <ListItem>
                <p><strong>Name:</strong> {ep.name}</p>
                <p><strong>Air Date:  </strong> {ep.air_date}</p>
              </ListItem>
            </List>
          ))}
        </CharacterDetails>
        <ImageDetail>
          <Image src={data.image} alt={data.name}/>
          <Button onClick={() => alert('Available soon')}>Buy merchandise</Button>
        </ImageDetail>
      </Card>
    </Wrapper>
  )
}

Details.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    status: PropTypes.string,
    species: PropTypes.string,
    type: PropTypes.string,
    gender: PropTypes.string,
    origin: PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
    }),
    location: PropTypes.shape({
      name: PropTypes.string,
      url: PropTypes.string,
    }),
    image: PropTypes.string,
    episode: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      air_date: PropTypes.string,
    })),
    url: PropTypes.string,
    created: PropTypes.string,
  }),
}

export default Details;
