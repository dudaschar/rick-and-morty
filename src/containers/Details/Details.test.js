import React from 'react';
import { render, screen } from 'src/testUtils/testUtils';

import Details from './Details';

const DATA = {
  "id": 2,
  "name": "Morty Smith",
  "status": "Alive",
  "species": "Human",
  "type": "",
  "gender": "Male",
  "origin": {
    "name": "Earth",
    "url": "https://rickandmortyapi.com/api/location/1"
  },
  "location": {
    "name": "Earth",
    "url": "https://rickandmortyapi.com/api/location/20"
  },
  "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
  "episode": [
    {
      name: 'Episode 1',
      air_date: '13 jun'
    }
  ],
  "url": "https://rickandmortyapi.com/api/character/2",
  "created": "2017-11-04T18:50:21.651Z"
}

describe('Details Container', () => {
  it('should render the all the character details', () => {
    const { asFragment } = render(
      <Details data={DATA} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a buy merchandise button', () => {
    render(<Details data={DATA} />);

    expect(screen.queryByText('Buy merchandise')).toBeTruthy();
  })
});
