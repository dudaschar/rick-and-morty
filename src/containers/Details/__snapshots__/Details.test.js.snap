// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Container should render the all the character details 1`] = `
<DocumentFragment>
  <div
    class="Wrapper-sc-1ev0bl-0 ddbBPU"
  >
    <div
      class="Wrapper__Card-sc-1ev0bl-2 gNmYJj"
    >
      <div
        class="Wrapper__CharacterDetails-sc-1ev0bl-1 dfCozP"
      >
        <h1
          class="Typography__CharacterName-sc-10u1p47-0 ixSsBh"
        >
          Morty Smith
        </h1>
        <p>
          <strong>
            Status:
          </strong>
           Alive
        </p>
        <p>
          <strong>
            Species:
          </strong>
           Human
        </p>
        <p>
          <strong>
            Gender:
          </strong>
           Male
        </p>
        <p>
          <strong>
            Type:
          </strong>
           N/A
        </p>
        <p>
          <strong>
            Origin:
          </strong>
           Earth
        </p>
        <p>
          <strong>
            Location:
          </strong>
           Earth
        </p>
        <h2
          class="Typography__EpisodesTitle-sc-10u1p47-1 hYXKTM"
        >
          Episodes
        </h2>
        <ul
          class="List-mqktuz-0 eAfkDN"
        >
          <li
            class="List__ListItem-mqktuz-1 ldSvgq"
          >
            <p>
              <strong>
                Name:
              </strong>
               Episode 1
            </p>
            <p>
              <strong>
                Air Date:  
              </strong>
               13 jun
            </p>
          </li>
        </ul>
      </div>
      <div
        class="Wrapper__ImageDetail-sc-1ev0bl-3 kbyIea"
      >
        <img
          alt="Morty Smith"
          class="Image-sc-1ma6ryq-0 kNpbEf"
          src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        />
        <button
          class="BaseButton-us6ig9-0 gNpRQk"
        >
          Buy merchandise
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
